src/wp-includes/class-wp-block.php

```diff
@@ -497,6 +461,51 @@ private function replace_html( string $block_content, string $attribute_name, $s
 		}
 	}
 
+	private static function get_block_bindings_processor( string $block_content ) {
+		$internal_processor_class = new class('', WP_HTML_Processor::CONSTRUCTOR_UNLOCK_CODE) extends WP_HTML_Processor {
+			/**
+			 * Replace the rich text content between a tag opener and matching closer.
+			 *
+			 * When stopped on a tag opener, replace the content enclosed by it and its
+			 * matching closer with the provided rich text.
+			 *
+			 * @param string $rich_text The rich text to replace the original content with.
+			 * @return bool True on success.
+			 */
+			public function replace_rich_text( $rich_text ) {
+				if ( $this->is_tag_closer() || ! $this->expects_closer() ) {
+					return false;
+				}
+
+				$depth = $this->get_current_depth();
+
+				$this->set_bookmark( '_wp_block_bindings_tag_opener' );
+				// The bookmark names are prefixed with `_` so the key below has an extra `_`.
+				$tag_opener = $this->bookmarks['__wp_block_bindings_tag_opener'];
+				$start      = $tag_opener->start + $tag_opener->length;
+				$this->release_bookmark( '_wp_block_bindings_tag_opener' );
```

Comment:
this is certainly welcome to do, to release the bookmarks, but performance-wise it’s not setting a _better_ example than leaving them dangling.

given that this function contains all of the scanning, it’s even fine to trap the bookmark bounds and re-use one, such as `set_bookmark( 'here' )`. I’ve considered exposing something like `->current_token()` but haven’t yet.